### ISSUE TITLE
upsample_linear1d_backward drops gradient terms when align_corners=True

### ISSUE THREAD

**Issue by truong-v** [FlagOpen/FlagGems#5218]
## Summary

The backward of a linear resize is a gather: each input pixel sums the gradients of every output pixel that interpolates from it. `upsample_linear1d_backward` does this by scanning a fixed window of output positions around a centre, and the host sizes that window from `out_w / in_w`.

That is the `align_corners=False` scale. With `align_corners=True` the input-to-output stretch is `(out_w - 1) / (in_w - 1)`, which is larger. Once the stretch exceeds the window, output positions that genuinely contribute fall outside the scanned range and their weight is silently dropped, so the gradient comes back short.

It needs a large ratio at a small input: about `out_w > 2 * in_w * in_w`. A 5x upsample of a length-2 signal is the cheapest way in. `align_corners=False` is unaffected.

(Unrelated to #2498, which is a tolerance flake in the forward op.)

The kernel was added in #2088.

## Reproduction

```python
import torch
import torch.nn.functional as F

import flag_gems

x = torch.randn(1, 1, 2, device="cuda", requires_grad=True)
g = torch.ones(1, 1, 10, device="cuda")

F.interpolate(x, size=10, mode="linear", align_corners=True).backward(g)
ref = x.grad.clone()
x.grad = None
with flag_gems.use_gems():
    F.interpolate(x, size=10, mode="linear", align_corners=True).backward(g)

print("eager:", ref.flatten().tolist())
print("gems :", x.grad.flatten().tolist())
```

```
eager: [5.0, 5.0]
gems : [4.888888835906982, 4.888888835906982]
```

## Environment

- FlagGems master (75cf3ea)
- NVIDIA L40S and H100 NVL
- torch 2.6.0+cu124, triton 3.2.0
